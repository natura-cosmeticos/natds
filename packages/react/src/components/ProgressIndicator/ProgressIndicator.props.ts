import { Size } from '@naturacosmeticos/natds-themes'

export type ProgressIndicatorSize = keyof Pick<Size, 'standard' | 'semi' | 'medium' | 'large'>
export interface ProgressIndicatorProps {
  /**
   * Describe the purpose of Progress Indicator
   * Read more [here](https://dequeuniversity.com/rules/axe/4.2/aria-progressbar-name?application=axeAPI)
   */
  ariaLabel: string

  /**
   * Show a layer behind the progress indicator
   * @default false
   */
  showLayer?: boolean

  /**
   * The Progress Indicator size
   * @default medium
   */
  size?: ProgressIndicatorSize

  /**
   * Optional ID for testing
   */
  testID?: string,
}
