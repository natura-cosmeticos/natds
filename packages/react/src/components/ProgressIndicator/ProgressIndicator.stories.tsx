/* eslint-disable max-len */
import React from 'react'
import { Story, Meta } from '@storybook/react'
import StoryContainer from '../../helpers/StoryContainer'
import { ProgressIndicator, ProgressIndicatorProps } from '.'

const componentStatus = `
---

**NOTE**: This component is available in the following variant:

  - ✅ \`circular\`
  - ❌ \`linear \`

With the following attribute status:

- **Size**
  - ✅  \`standard\`
  - ✅  \`semi\`
  - ✅  \`medium\`
  - ✅  \`large\`
- ✅ **Layer**

---
`

export default {
  title: 'Components/ProgressIndicator',
  component: ProgressIndicator,
  parameters: {
    componentSubtitle: 'Progress indicators express an unspecified wait time.',
    docs: { description: { component: componentStatus } }
  }
} as Meta

export const Playground: Story<ProgressIndicatorProps> = (args) => <ProgressIndicator {...args} />

export const Size: Story<ProgressIndicatorProps> = (args) => (
  <StoryContainer>
    <ProgressIndicator {...args} size="standard" />
    <ProgressIndicator {...args} size="semi" />
    <ProgressIndicator {...args} />
    <ProgressIndicator {...args} size="large" />
  </StoryContainer>
)
export const Layer: Story<ProgressIndicatorProps> = (args) => <ProgressIndicator {...args} />
Layer.args = { ...Playground.args, showLayer: true }
