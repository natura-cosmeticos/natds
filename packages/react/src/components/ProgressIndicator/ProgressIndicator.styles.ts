import { createUseStyles } from 'react-jss'
import { Theme } from '@naturacosmeticos/natds-themes'
import { ProgressIndicatorProps } from './ProgressIndicator.props'

const styles = createUseStyles((theme: Theme) => ({
  layer: {
    height: ({ size }: ProgressIndicatorProps) => size && theme.size[size] + 8,
    width: ({ size }: ProgressIndicatorProps) => size && theme.size[size] + 8,
    backgroundColor: ({ showLayer }: ProgressIndicatorProps) => showLayer && theme.color.surface,
    borderRadius: '50%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    padding: theme.size.micro
  },
  spinner: {
    height: ({ size }: ProgressIndicatorProps) => size && theme.size[size],
    width: ({ size }: ProgressIndicatorProps) => size && theme.size[size],
    border: `${theme.size.micro}px solid ${theme.color.primary}`,
    borderLeftColor: 'transparent',
    borderTopColor: 'transparent',
    borderRadius: '50%',
    animation: '$spin 1.7s linear infinite'
  },
  '@keyframes spin': {
    '0%': {
      transform: 'rotate(0deg)'
    },
    '75%': {
      transform: 'rotate(360deg)'
    },
    '100%': {
      transform: 'rotate(720deg)'
    }
  }
}))

export default styles
