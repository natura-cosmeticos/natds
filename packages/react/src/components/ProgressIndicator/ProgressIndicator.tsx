import React from 'react'
import { ProgressIndicatorProps } from './ProgressIndicator.props'
import styles from './ProgressIndicator.styles'

const ProgressIndicator = ({
  showLayer = false,
  size = 'medium',
  testID = 'ds-progress-indicator',
  variant = 'circular'
}: ProgressIndicatorProps): JSX.Element => {
  const { base, layer } = styles({ size, showLayer, variant })

  return (
    <div className={layer} role="progressbar" data-testid={testID}>
      <div className={base} />
    </div>
  )
}

export default ProgressIndicator
