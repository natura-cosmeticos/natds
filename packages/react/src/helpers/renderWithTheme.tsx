import React from 'react'
import { JssProvider, SheetsRegistry } from 'react-jss'
import { render } from '@testing-library/react'
import { ThemeProvider } from '../ThemeProvider'

const renderWithTheme = (component) => {
  const sheets = new SheetsRegistry()

  const renderTree = render(
    <JssProvider registry={sheets}>
      <ThemeProvider>
        {component}
      </ThemeProvider>
    </JssProvider>
  )

  return { styles: sheets.toString(), component: renderTree }
}

export default renderWithTheme
