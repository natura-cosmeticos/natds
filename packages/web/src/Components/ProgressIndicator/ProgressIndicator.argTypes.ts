/* eslint-disable no-magic-numbers */
export const argTypes = {
  className: {
    control: { type: null },
    description: "Override or extend the styles applied to the component.",
    table: { type: { summary: "object" } },
  },
  color: {
    control: {
      type: "select",
      options: ["primary", "secondary", "inherit"],
    },
    description:
      "The color of the component. It supports those theme colors that make sense for this component.",
  },
  size: {
    control: {
      type: "select",
      options: [24, 32, 40, 48],
    },
    description: "The size of the circle.",
    table: { type: { summary: "number" } },
  },
  disableShrink: {
    table: {
      defaultValue: { summary: false },
      type: { summary: "bool" },
    },
    description: "If true, the shrink animation is disabled.",
  },
};

export default argTypes;
