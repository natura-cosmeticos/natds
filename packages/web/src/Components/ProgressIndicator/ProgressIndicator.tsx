/* eslint-disable no-magic-numbers */
/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import MaterialCircularProgress from "@material-ui/core/CircularProgress";
import { IProgressIndicatorProps } from "./ProgressIndicator.props";
import { useStyles } from "./ProgressIndicator.styles";

/**
 * ```
 * import { ProgressIndicator } from "@naturacosmeticos/natds-web"
 * ```
 */

export const ProgressIndicator = ({ layer = false, size = 48, ...props }: IProgressIndicatorProps) => {
  const { circularLayer } = useStyles(props);

  return (
    <>
      {layer
        ? <div className={circularLayer}>
          <MaterialCircularProgress {...props} size={size} />
        </div>
        : <MaterialCircularProgress {...props} size={size}/>
      }
    </>
  );
};

ProgressIndicator.displayName = "ProgressIndicator";

/**
 * @deprecated `CircularProgress` component will be deprecated soon.
 * Use `ProgressIndicator` instead
 */
export const CircularProgress = ProgressIndicator;

export default { CircularProgress, ProgressIndicator };
