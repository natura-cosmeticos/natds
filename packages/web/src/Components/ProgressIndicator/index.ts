import { CircularProgress, ProgressIndicator } from './ProgressIndicator'

export { IProgressIndicatorProps } from './ProgressIndicator.props'
export { CircularProgress, ProgressIndicator }
export default ProgressIndicator
