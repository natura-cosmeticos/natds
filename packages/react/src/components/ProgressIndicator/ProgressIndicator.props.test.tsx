import React from 'react'
import renderWithTheme from '../../helpers/renderWithTheme'
import ProgressIndicator from './ProgressIndicator'

describe('Progress Indicator', () => {
  it('should render correctly with default props', () => {
    const { styles, component } = renderWithTheme(<ProgressIndicator />)

    expect([styles.toString(), component.container]).toMatchSnapshot()
  })
})
