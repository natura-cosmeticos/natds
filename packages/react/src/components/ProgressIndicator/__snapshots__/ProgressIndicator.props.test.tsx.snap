// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Indicator should render correctly with default props 1`] = `
Array [
  ".layer-0-2-1 {
  display: flex;
  padding: 4px;
  align-items: center;
  border-radius: 50%;
  justify-content: center;
}
.spinner-0-2-2 {
  border: 4px solid #f4ab34;
  animation: keyframes-spin-0-2-3 1.7s linear infinite;
  border-radius: 50%;
  border-top-color: transparent;
  border-left-color: transparent;
}
@keyframes keyframes-spin-0-2-3 {
  0% {
    transform: rotate(0deg);
  }
  75% {
    transform: rotate(360deg);
  }
  100% {
    transform: rotate(720deg);
  }
}
  .layer-d0-0-2-4 {
    height: 56px;
    width: 56px;
  }
  .spinner-d1-0-2-5 {
    height: 48px;
    width: 48px;
  }",
  <div>
    <div
      class="layer-0-2-1 layer-d0-0-2-4"
      data-testid="ds-progress-indicator"
      role="progressbar"
    >
      <div
        class="spinner-0-2-2 spinner-d1-0-2-5"
      />
    </div>
  </div>,
]
`;
